features/step_definitions/command_steps.rb

```diff
@@ -217,20 +217,21 @@ def checkRestart(host, time_out)
 end
 
 When(/^I run "([^"]*)" on "([^"]*)"$/) do |cmd, target|
-  if target == "server"
+  case target
+  when "server"
     _out, $fail_code = $server.run(cmd, false)
-  elsif target == "ceos-minion"
-    _out, _code = $ceos_minion.run(cmd, false)
-  elsif target == "ssh-minion"
-    _out, _code = $ssh_minion.run(cmd, false)
-  elsif target == "sle-minion"
-    _out, _code = $minion.run(cmd, false)
-  elsif target == "sle-client"
-    _out, _code = $client.run(cmd, false)
+  when "ceos-minion"
+    _out, $fail_code = $ceos_minion.run(cmd, false)
+  when "ssh-minion"
+    _out, $fail_code = $ssh_minion.run(cmd, false)
+  when "sle-minion"
+    _out, $fail_code = $minion.run(cmd, false)
+  when "sle-client"
+    _out, $fail_code = $client.run(cmd, false)
```

Comment:
we can add here a else statement 

``` else 
 raise "no valid target given"```
for be in the safe side.

It look goods for me, just rebase then after the 3 commits in one